python/src/dataset.rs

```diff
@@ -378,39 +378,30 @@ impl Dataset {
         }
     }
 
-    fn create_version_from_fragments(
-        self_: PyRef<Self>,
+    #[staticmethod]
+    fn commit(
+        dataset_uri: &str,
         schema: &PyAny,
         fragments: Vec<&PyAny>,
     ) -> PyResult<Self> {
-        let schema = ArrowSchema::from_pyarrow(schema)?;
+        let rt = Arc::new(Runtime::new()?);
+        let arrow_schema = ArrowSchema::from_pyarrow(schema)?;
+        let schema = Schema::try_from(&arrow_schema).map_err(|e| {
+            PyValueError::new_err(format!("Failed to convert Arrow schema to Lance schema: {}", e))
+        })?;
         let fragment_metadata = fragments
             .iter()
             .map(|f| f.extract::<FragmentMetadata>().map(|fm| fm.inner))
             .collect::<PyResult<Vec<Fragment>>>()?;
-
-        let dataset_schema = self_.ds.schema();
-        let new_schema = Schema::try_from(&schema)
-            .map_err(|e| PyValueError::new_err(format!("Can not convert schema to lance: {}", e)))?;
-        let added_on_schema = new_schema
-            .exclude(dataset_schema)
-            .map_err(|e| PyValueError::new_err(format!("Can not convert schema to lance: {}", e)))?;
-        let new_schema_with_id = dataset_schema
-            .merge(&added_on_schema)
-            .map_err(|e| PyValueError::new_err(format!("Can not convert schema to lance: {}", e)))?;
-        let ds = self_
-            .rt
+        let ds = rt
             .block_on(async {
-                self_
-                    .ds
-                    .create_version_from_fragments(&new_schema_with_id, &fragment_metadata)
-                    .await
+                LanceDataset::commit(dataset_uri, &schema, &fragment_metadata).await
```

Comment:
is commit the right terminology here?

Reply:
So from the DB term, writing all fragments are the preparing phase of a transaction, and this last step "commit" the change to the dataset (as making them visible)?